mesonbuild/mcompile.py

```diff
@@ -137,7 +137,7 @@ def generate_target_names_ninja(target: ParsedTargetName, builddir: Path, intros
     else:
         return [str(Path(out_file).relative_to(builddir.resolve())) for out_file in intro_target['filename']]
 
-def get_parsed_args_ninja(options: 'argparse.Namespace', builddir: Path) -> T.List[str]:
+def get_parsed_args_ninja(options: 'argparse.Namespace', builddir: Path) -> T.Tuple[T.List[str], T.Dict[str, str]]:
```

Comment:
Looking at it again, this should actually be `T.Tuple[T.List[str], T.Optional[T.Dict[str, str]]]`, because `env` can be `None`. Same for the other uses of `env` below. I am surprised that mypy doesn't warn about this, though. @dcbaker what do you think?

Reply:
Yeah, this should either return `cmd, {}`, or the Dict needs to be optional. I'm surprised that mypy isn't warning about this too..

Reply:
If it were my I'd probably just return an empty dict you the calling code doesn't need to handle the None case.

Reply:
Looking at the code, this happens to work because the receiving code takes an `Optional[Dict[str, str]]`, so either way I guess, but there will be less code to rework if you just always return `{}` instead of None

Reply:
The two are not equivalent. In Python, when you set a kwarg to `None` that's considered to be unset (by convention), so `env=None` means "default: inherit process env" while `env={}` means "use an empty env". `env={}` will break everything.

Reply:
I've pushed a commit that fixes this, and also rebased it on top of https://github.com/mesonbuild/meson/pull/7730.